src/pymodaq/utils/parameter/ioxml.py

```diff
@@ -348,9 +350,12 @@ def parameter_to_xml_file(param, filename: Union[str, Path]):
     fname = parent.joinpath(filename + ".xml")  # forcing the right extension on the filename
     xml_elt = walk_parameters_to_xml(param=param)
     tree = ET.ElementTree(xml_elt)
+    if not overwrite and os.path.isfile(str(fname)):
```

Comment:
fname is a Path object so you should use the corresponding method :
```
fname.exists() and fname.is_file()
```

Reply:
do i really need to check if it exists and if it is a file ?
i don't think it can be a file if it doesn't exit.

Reply:
in python if the first clause of an *and* fails it doesn't test the second. But it can exist and be a directory... (unlikely I agree)

Reply:
I agree those 3 conditions meet the right logic. 
I just feel that not overwrite and fname.is_file() would still produce the same result as it can't be a file if it doesn't exist so is_file() is also a test of its existence, but i am just spliting hairs :)